synapse/storage/events.py

```diff
@@ -2091,11 +2096,16 @@ def _delete_old_state_txn(self, txn, room_id, topological_ordering):
                 retcols=["state_group"],
                 keyvalues={},
             )
-            new_state_edges.extend(row["state_group"] for row in rows)
+            new_state_edges.extend(
+                row["state_group"] for row in rows
+                if row["state_group"] not in state_groups_to_delete
```

Comment:
Can you comment this please? It took me a few moments to realise what was going on